app/workers/AddJsonToMessagesTransformer.scala

```diff
@@ -45,38 +46,15 @@ private[workers] class AddJsonToMessagesTransformer @Inject()(
 
   private val settings = workerConfig.addJsonToMessagesWorkerSettings
 
-  private val supervisionStrategy: Attributes = ActorAttributes.supervisionStrategy {
-    case WorkerResumeableException(message, cause) =>
-      logger.warn(message, cause)
-      Supervision.resume
-
-    case WorkerResumeable(message) =>
-      logger.info(message)
-      Supervision.resume
-
-    case WorkerUnresumeable(message) =>
-      logger.error(message)
-      Supervision.stop
-
-    case WorkerUnresumeableException(message, cause) =>
-      logger.error(message, cause)
-      Supervision.stop
-
-    case NonFatal(e) =>
-      logger.warn("Worker saw an exception but will resume", e)
-      Supervision.resume
-
-    case e =>
-      logger.error("Worker saw a fatal exception and will be stopped", e)
-      Supervision.stop
-  }
+  private val supervisionStrategy: WorkerProcessingProblemSupervisionStrategyProvider =
```

Comment:
Would changing the name of this val as well let us avoid having `supervisionStrategy.supervisionStrategy` later?

Reply:
Done. I've renamed the `WorkerProcessingProblemSupervisionStrategyProvider` to `WorkerSupervisionDeciderProvider` and the variable names to `supervisionDeciderProvider`.